internal/synclet/client_adapter.go

```diff
@@ -82,17 +83,26 @@ func (s *SyncletCli) UpdateContainer(
 }
 
 func (s *SyncletCli) ContainerIDForPod(ctx context.Context, podID k8s.PodID, imageID reference.NamedTagged) (cID k8s.ContainerID, err error) {
-	start := time.Now()
-	for time.Since(start) < containerIdTimeout {
+	timeout := time.NewTimer(containerIdTimeout)
```

Comment:
this uh. seems more complicated than it needs to be? all you need is `time.Sleep(retryDelay)` before calling the func again, and `if ctx.Err() != nil { return "", ctx.Err() }`

Reply:
My understanding is that the idiomatic way to do things is to never actually Sleep, with pauses in loops always on some form of `select`, so that we're always ready to abort in event of timeout or cancelled context. @nicks does this look right to you?

Reply:
ya, tho i think i'd usually be more idiomatic to use time.After()